mypyc/ops_misc.py

```diff
@@ -164,3 +164,14 @@ def emit_isinstance(emitter: EmitterInterface, args: List[str], dest: str) -> No
         result_type=bool_rprimitive,
         error_kind=ERR_MAGIC,
         emit=emit_isinstance)
+
+
+# Faster isinstance() that only works with native classes and doesn't perform type checking
+# of the type argument.
+fast_isinstance_op = func_op(
+    'builtins.isinstance',
+    arg_types=[object_rprimitive, object_rprimitive],
+    result_type=bool_rprimitive,
+    error_kind=ERR_NEVER,
+    emit=simple_emit('{dest} = PyObject_TypeCheck({args[0]}, (PyTypeObject *){args[1]});'),
+    priority=0)
```

Comment:
What all is different about `TypeCheck` and `IsInstance`?

Reply:
> What all is different about TypeCheck and IsInstance?

Performance. The former skips some dances with `__instancecheck__`.